examples/molecular_dynamics/lennard_jones_system.py

```diff
@@ -0,0 +1,280 @@
+        num_layers_edge_processor=cfg.model.num_layers_edge_processor,
+        num_layers_node_encoder=None,  # No node encoder
+        num_layers_node_decoder=cfg.model.num_layers_node_decoder,
+        hidden_dim_edge_encoder=cfg.model.hidden_dim_edge_encoder,
+    ).to(dist.device)
+
+    if dist.distributed:
+        ddps = torch.cuda.Stream()
+        with torch.cuda.stream(ddps):
+            model = DistributedDataParallel(
+                model,
+                device_ids=[dist.local_rank],
+                output_device=dist.device,
+                broadcast_buffers=dist.broadcast_buffers,
+                find_unused_parameters=dist.find_unused_parameters,
+            )
+        torch.cuda.current_stream().wait_stream(ddps)
+
+    optimizer = torch.optim.Adam(model.parameters(), lr=cfg.lr.start_lr)
+
+    scheduler = torch.optim.lr_scheduler.ExponentialLR(optimizer, gamma=cfg.lr.gamma)
+
+    LaunchLogger.initialize(use_mlflow=True)
+
+    # define constants
+    distance_threshold = cfg.distance_threshold
+    box_size = cfg.box_size
+    force_mean, force_sd = compute_mean_var(
+        to_absolute_path(os.path.join("./", "lj_data"))
+    )
+
+    # Attempt to load latest checkpoint if one exists
+    loaded_epoch = load_checkpoint(
+        "./checkpoints",
+        models=model,
+        optimizer=optimizer,
+        scheduler=scheduler,
+        device=dist.device,
+    )
+
+    for epoch in range(max(1, loaded_epoch + 1), cfg.max_epochs + 1):
+        with LaunchLogger(
+            "train", epoch=epoch, num_mini_batch=len(dataloader), epoch_alert_freq=1
+        ) as log:
+            model.train()
+            for data in dataloader:
+                # Create edges
+                pos = data[0][
+                    0
+                ]  # Select first element of the list (works only for batchsize 1)
+                forces = data[1][0]
+                r = get_rotation_matrix()
+
+                pos, forces = prepare_input(
+                    pos, forces, box_size, r, add_random_noise=True
+                )
+                src, dst, edge_features = create_edges(
+                    pos, distance_threshold, box_size
+                )
+                g = dgl.graph((src, dst)).to(dist.device)
+
+                node_fea = torch.ones(
+                    size=(pos.shape[0], cfg.model.hidden_dim_edge_encoder)
+                ).to(dist.device)
+                edge_fea = (
+                    torch.tensor(np.array(edge_features), dtype=torch.float32)
+                    .view(-1, 4)
+                    .to(dist.device)
+                )
+
+                out = model(node_fea, edge_fea, g)
+                true_out = torch.tensor(
+                    (forces - force_mean) / force_sd, dtype=torch.float32
+                ).to(dist.device)
+
+                optimizer.zero_grad()
+                loss = F.l1_loss(out, true_out) + 0.001 * torch.mean(out).abs()
```

Comment:
Any commentary on this loss function? I think it would be helpful to the user to point out the origin/interpretation of this loss.

Reply:
Thanks, added!